### PR TITLE
Feature#6 create register of checking

### PR DESCRIPTION
**Pull request by oscaregr**
# Description

What did you implemented/Why did you implemented this:

we created the model for checks.
we conected employee_id like secundary_key for checks
we implement create one check from one employee from the index view

## Testing
not yet

## Screenshots
image
image

### CODE REVIEW

**Review by MaxRetana — APPROVED** [review]
Epic

File: app/assets/stylesheets/views/_checkView.scss [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+		&:hover {
+			background-color: #b4f5b8;
+		}
+	}
+	&__off {
+		background-color: $--check-off-color;
+		border: 0;
+		color: $--black-color;
+		&:hover {
+			background-color: #e58581;
```
Comment (JuanequeX) [inline app/assets/stylesheets/views/_checkView.scss:37]: Remove this color for a variable

File: app/assets/stylesheets/views/_checkView.scss [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+	justify-content: space-between;
+	&__in {
+		border: 0;
+		background-color: $--check-in-color;
+		color: $--black-color;
+		&:hover {
+			background-color: #b4f5b8;
+		}
+	}
+	&__off {
```
Comment (JuanequeX) [inline app/assets/stylesheets/views/_checkView.scss:32]: add a blank line in the line `32`

File: app/models/employee.rb [hunk 3]
```diff
@@ -4,5 +4,7 @@ class Employee < ApplicationRecord
 	validates :name, presence: true
 	validates :position, presence: true
 	validates :number_private, presence: true
-	validates :is_employee, presence: true
+	validates :is_employee, presence: false
+	# self.primary_key = "employee"
```
Comment (juperameza) [inline app/models/employee.rb:8]: If you don need it delete it

File: app/assets/stylesheets/views/_checkView.scss [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+		text-shadow: 0.5px 1px 1px $--shadow-text-color;
+	}
+
+	.alert {
+		color: $--check-off-color,
+	}
+	.notice {
+		color: $--check-in-color;
+	}
+}
```
Comment (JuanequeX) [inline app/assets/stylesheets/views/_checkView.scss:64]: Add a blank line here!

**Review by Reiyirak — APPROVED** [review]
LGTM, maybe an empty line at the end for _checkView.scss and _variables.scss

File: app/views/checks/index.html.erb [hunk 5]
```diff
@@ -26,8 +26,8 @@
 			<% end %>
 		</div>
 		<div class="check__buttons_container">
-			<button type="submit" id='checkin' class="btn btn-primary check__buttons_container_in">Check In</button>
-			<button type="submit" id='checkoff' class="btn btn-primary check__buttons_container_off">Check Off</button>
+			<button type="submit" name="check" value="in" id='checkin' class="btn btn-primary check__buttons_container__in">Check In</button>
```
Comment (juperameza) [inline app/views/checks/index.html.erb:29]: Why are you using html buttons instead of embedded ruby ??

**Review by juperameza — CHANGES_REQUESTED** [review]

File: app/controllers/checks_controller.rb [hunk 6]
```diff
@@ -7,15 +7,24 @@ def index
   end
 
   def create
-    check = Employee.find_by(check_params)
-    check.nil? ? flash[:notice] = 'Successfully check' : flash[:alert] = 'User no avalible'
+    @employee = Employee.find_by(employee_params)
+    if @employee != nil
```
Comment (JuanequeX) [inline app/controllers/checks_controller.rb:11]: I think is better use `@employee.nil?`

**Review by JuanequeX — CHANGES_REQUESTED** [review]

**Review by juperameza — APPROVED** [review]
Lgtm uwu

File: app/assets/stylesheets/views/_checkView.scss [hunk 7]
```diff
@@ -0,0 +1,77 @@
+@import 'utilities/variables';
+
+.content {
+
```
Comment (JuanequeX) [inline app/assets/stylesheets/views/_checkView.scss:4]: Delete this line

File: app/assets/stylesheets/views/_checkView.scss [hunk 8]
```diff
@@ -0,0 +1,77 @@
+@import 'utilities/variables';
```
Comment (JuanequeX) [inline app/assets/stylesheets/views/_checkView.scss:1]: this import is not necessary if declare inside of the `app.scss` file

File: app/assets/stylesheets/views/_checkView.scss [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+	display: flex;
+	justify-content: center;
+	margin: 3rem 25rem;
+	padding: 3rem;
+}
+
+.time__container {
+
+	text-align: center;
+	font-family: sans-serif;
```
Comment (JuanequeX) [inline app/assets/stylesheets/views/_checkView.scss:16]: create a global variable for fonts

File: app/assets/stylesheets/views/_checkView.scss [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+		&:hover {
+			background-color: $--green-check;
+		}
+	}
+
+	&__off {
+		background-color: $--check-off-color;
+		border: 0;
+		color: $--black-color;
+		border-radius: 0.3rem;
```
Comment (JuanequeX) [inline app/assets/stylesheets/views/_checkView.scss:44]: sort alphabetically this block

**Review by JuanequeX — CHANGES_REQUESTED** [review]

File: app/assets/stylesheets/views/_checkView.scss (hunk 9 above)
Comment (xManuel007) [reply app/assets/stylesheets/views/_checkView.scss:16]: done

File: app/assets/stylesheets/views/_checkView.scss (hunk 8 above)
Comment (xManuel007) [reply app/assets/stylesheets/views/_checkView.scss:1]: done

File: app/assets/stylesheets/views/_checkView.scss [hunk 11]
```diff
@@ -0,0 +1,72 @@
...
+	border-radius: 2rem;
+	display: flex;
+	justify-content: center;
+	margin: 3rem 25rem;
+	padding: 3rem;
+}
+
+.time__container {
+	text-align: center;
+	font-family: $--primary-font;
```
Comment (JuanequeX) [inline app/assets/stylesheets/views/_checkView.scss:12]: sort alphabetically this block

File: app/assets/stylesheets/views/_checkView.scss [hunk 12]
```diff
@@ -0,0 +1,72 @@
...
+		font-size: 7rem;
+	}
+}
+
+.check__buttons_container {
+	display: flex;
+	margin-top: 1rem;
+	justify-content: space-around;
+	padding: 0rem 1.5rem 0rem 1.5rem;
+
```
Comment (JuanequeX) [inline app/assets/stylesheets/views/_checkView.scss:24]: sort alphabetically this block

File: app/assets/stylesheets/views/_checkView.scss [hunk 13]
```diff
@@ -0,0 +1,72 @@
...
+	&__in {
+		border: 0;
+		background-color: $--check-in-color;
+		color: $--black-color;
+		border-radius: 0.3rem;
```
Comment (JuanequeX) [inline app/assets/stylesheets/views/_checkView.scss:29]: sort alphabetically this block

**Review by JuanequeX — CHANGES_REQUESTED** [review]

**Review by JuanequeX — APPROVED** [review]
LGTM

**Pull request merged**